### PR TITLE
DAOS-5748 test: fix parsing of dmg pool create svc replicas

### PR DESCRIPTION
**Pull request by kccain**
Change the regex used to parse the dmg command standard output
to gather the pool service replica ranks and construct an array
that is used in the pydaos class DaosPool API implementation.
Before this change, only the first pool service replica rank
was parsed. With this change, all replica ranks are parsed from
the comma-separated list output.

Signed-off-by: Kenneth Cain <kenneth.c.cain@intel.com>

### CODE REVIEW

**Comment by kccain** [comment]
Functional Leap15 failures are being seen on master and by a number of engineers on PRs. Is being investigated by team currently.

This [macsio timeout ](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-3572/2/testReport/junit/(root)/MacsioTest/Test___Functional_Hardware_Large___4___io_macsio_test_py_MacsioTest_test_macsio_daos_vol_container_dfuse_hosts_macsio_openmpi_pool_servers_4e26/) is associated with [DAOS-5591](https://jira.hpdd.intel.com/browse/DAOS-5591)

This [certificate error in csum error logging test](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-3572/2/testReport/junit/(root)/CSumErrorLog/Test___Functional_Hardware_Medium___1___checksum_csum_error_logging_py_CSumErrorLog_test_csum_error_logging_test_z_num_clients_num_replicas_hosts_servers_761a/) is associated with known issue [DAOS-5745](https://jira.hpdd.intel.com/browse/DAOS-5745)

**Review by amjustin13 — APPROVED** [review]
LGTM

**Comment by kccain** [comment]
Build 3 failures

[DaosCoreTestRebuild/Test___Functional_Hardware_Medium___01___daos_test_daos_core_test_rebuild_py_DaosCoreTestRebuild_test_rebuild_transport_config_test_r_0_10_num_clients_num_replicas_transport_config_hosts_pool_0_1_transport_config_5153](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-3572/3/testReport/junit/(root)/DaosCoreTestRebuild/Test___Functional_Hardware_Medium___01___daos_test_daos_core_test_rebuild_py_DaosCoreTestRebuild_test_rebuild_transport_config_test_r_0_10_num_clients_num_replicas_transport_config_hosts_pool_0_1_transport_config_5153/)
associated with [DAOS-5784](https://jira.hpdd.intel.com/browse/DAOS-5784)

[CSumErrorLog/Test___Functional_Hardware_Medium___1___checksum_csum_error_logging_py_CSumErrorLog_test_csum_error_logging_test_z_num_clients_num_replicas_hosts_servers_dda0](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-3572/3/testReport/junit/(root)/CSumErrorLog/Test___Functional_Hardware_Medium___1___checksum_csum_error_logging_py_CSumErrorLog_test_csum_error_logging_test_z_num_clients_num_replicas_hosts_servers_dda0/)
associated with [DAOS-5745](https://jira.hpdd.intel.com/browse/DAOS-5745)

[OSAOnlineDrain/Test___Functional_Hardware_Large___1___osa_osa_online_drain_py_OSAOnlineDrain_test_osa_online_drain_container_hosts_clientslots_iorflags_pool_servers_server_manager_a21e](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-3572/3/testReport/junit/(root)/OSAOnlineDrain/Test___Functional_Hardware_Large___1___osa_osa_online_drain_py_OSAOnlineDrain_test_osa_online_drain_container_hosts_clientslots_iorflags_pool_servers_server_manager_a21e/)
possibly associated with previously closed ticket [DAOS-5538](https://jira.hpdd.intel.com/browse/DAOS-5538)

**Comment by kccain** [comment]
There has been some discussion by @mjmac that the class might execute dmg with the --json option to avoid issues with regex parsing of that tool standard output (here I guess for pool create, but not sure how such a change might expand beyond pool create usages). @phender any thoughts on that?

**Comment by mjmac** [comment]
> 
> 
> There has been some discussion by @mjmac that the class might execute dmg with the --json option to avoid issues with regex parsing of that tool standard output (here I guess for pool create, but not sure how such a change might expand beyond pool create usages). @phender any thoughts on that?

If this PR works OK for now, I'm not suggesting that it should be expanded to encompass the update to using json output. I was just offering it as a possibility if the regex approach continued to be unreliable.

**Comment by phender** [comment]
> There has been some discussion by @mjmac that the class might execute dmg with the --json option to avoid issues with regex parsing of that tool standard output (here I guess for pool create, but not sure how such a change might expand beyond pool create usages). @phender any thoughts on that?

So we have the option to enable dmg json output easily in each [test yaml file](https://github.com/daos-stack/daos/blob/master/src/tests/ftest/server/dynamic_start_stop.yaml#L17) (or we could change the default in [dmg_utils_base.py](https://github.com/daos-stack/daos/blob/master/src/tests/ftest/util/dmg_utils_base.py#L56)).  The issue in either case is the use of json output would continue for any dmg command issued during the test and it would require updating all the helper functions in ftest/util/dmg_utils.py.

In terms of the amount of work, isn't it simpler just to fix the regex?
https://github.com/daos-stack/daos/blob/master/src/tests/ftest/util/dmg_utils.py#L436
```
r"Service replicas:\s+([A-Za-z0-9-]+)",
```
```
r"Service replicas:\s+([A-Za-z0-9-,]+)",
```
and
https://github.com/daos-stack/daos/blob/master/src/tests/ftest/util/dmg_utils.py#L440
```
data["svc"] = match[0][1]
```
```
data["svc"] = match[0][1].split(",")
```

File: src/tests/ftest/util/dmg_utils.py [hunk 1]
```diff
@@ -433,7 +433,7 @@ def pool_create(self, scm_size, uid=None, gid=None, nvme_size=None,
         else:
             match = re.findall(
                 r"UUID:\s+([A-Za-z0-9-]+),\s+"
-                r"Service replicas:\s+([A-Za-z0-9-]+)",
+                r"Service replicas:\s+([0-9]+(,[0-9]+)*)",
```
Comment (phender) [inline src/tests/ftest/util/dmg_utils.py:436]: To capture the entire list, e.g. "1,2,3", you'll want to use this pattern:
```suggestion
                r"Service replicas:\s+([A-Za-z0-9-,]+)",
```

**Review by phender — APPROVED** [review]
LGTM

**Pull request merged**